include/deal.II/grid/manifold.h

```diff
@@ -396,14 +456,14 @@ class Manifold : public Subscriptor
    * by implementing only the get_new_point() function.
    *
    * The implementation does not allow for @p surrounding_points and
-   * @p new_points to point to the same vector, so make sure to pass different
+   * @p new_points to point to the same array, so make sure to pass different
    * objects into the function.
    */
   virtual
   void
-  add_new_points (const std::vector<Point<spacedim> > &surrounding_points,
-                  const Table<2,double>               &weights,
-                  std::vector<Point<spacedim> >       &new_points) const;
+  add_new_points (const ArrayView<const Point<spacedim>> &surrounding_points,
+                  const Table<2,double>                  &weights,
+                  ArrayView<Point<spacedim>>              new_points) const;
```

Comment:
The name `add_new_points` for this function is now no longer accurate. It used to be that the new points were *added* to a vector, but now they're just written into. Consequently, I think the function should just be called `get_new_points()`.

The function documentation should also say, somewhere, that the number of elements in `new_points` should equal the number of rows in `weights`.

Reply:
I think you mean the number of columns in weights :)

I agree with the name change. Its just one more incompatibility on top of the pile of incompatibilities at this point; we just added this function in 8.5 so I think we will be okay (and anyone who is a hardcore `Manifold` user will have to rewrite their codes anyway).

Reply:
Clarification: the documentation already states that `The number of columns in @p weights must match the length of @p new_points`.

Reply:
Unless you've already done it, we can also defer the renaming of `add_new_points()` to a later patch.

Reply:
I went ahead and changed it anyway. I'll push changes in a few more minutes.